src/Database/Esqueleto/Experimental.hs

```diff
@@ -1205,52 +1205,47 @@ withRecursive baseCase unionKind recursiveCase = do
   Q $ W.tell mempty{sdCteClause = [clause]}
   pure refFrom
 
-type family ToAliasT a where
-  ToAliasT (SqlExpr (Value a)) = SqlExpr (Value a)
-  ToAliasT (SqlExpr (Entity a)) = SqlExpr (Entity a)
-  ToAliasT (SqlExpr (Maybe (Entity a))) = SqlExpr (Maybe (Entity a))
-  ToAliasT (a, b) = (ToAliasT a, ToAliasT b)
-  ToAliasT (a, b, c) = (ToAliasT a, ToAliasT b, ToAliasT c)
-  ToAliasT (a, b, c, d) = (ToAliasT a, ToAliasT b, ToAliasT c, ToAliasT d)
-  ToAliasT (a, b, c, d, e) = (ToAliasT a, ToAliasT b, ToAliasT c, ToAliasT d, ToAliasT e)
-  ToAliasT (a, b, c, d, e, f) = (ToAliasT a, ToAliasT b, ToAliasT c, ToAliasT d, ToAliasT e, ToAliasT f)
-  ToAliasT (a, b, c, d, e, f, g) = (ToAliasT a, ToAliasT b, ToAliasT c, ToAliasT d, ToAliasT e, ToAliasT f, ToAliasT g)
-  ToAliasT (a, b, c, d, e, f, g, h) = (ToAliasT a, ToAliasT b, ToAliasT c, ToAliasT d, ToAliasT e, ToAliasT f, ToAliasT g, ToAliasT h)
-
 -- Tedious tuple magic
 class ToAlias a where
+  type ToAliasT a :: *
   toAlias :: a -> SqlQuery (ToAliasT a)
 
 instance ToAlias (SqlExpr (Value a)) where
+  type ToAliasT (SqlExpr (Value a)) = SqlExpr (Value a)
   toAlias v@(EAliasedValue _ _) = pure v
   toAlias v = do
     ident <- newIdentFor (DBName "v")
     pure $ EAliasedValue ident v
 
 instance ToAlias (SqlExpr (Entity a)) where
+  type ToAliasT (SqlExpr (Entity a)) = SqlExpr (Entity a)
   toAlias v@(EAliasedEntityReference _ _) = pure v
   toAlias v@(EAliasedEntity _ _) = pure v
   toAlias (EEntity tableIdent) = do
     ident <- newIdentFor (DBName "v")
     pure $ EAliasedEntity ident tableIdent
 
 instance ToAlias (SqlExpr (Maybe (Entity a))) where
+  type ToAliasT (SqlExpr (Maybe (Entity a))) = SqlExpr (Maybe (Entity a))
   toAlias (EMaybe e) = EMaybe <$> toAlias e
 
 instance (ToAlias a, ToAlias b) => ToAlias (a,b) where
+  type ToAliasT (a, b) = (ToAliasT a, ToAliasT b)
```

Comment:
So this equation is *not* the identity - it's pushing the `ToAliasT` to members of the tuple.

Reply:
Right, sleepy brain couldn't remember why this type family existed. It's the unfortunate requirement of distributive tuple magic.

Reply:
OK, I'm now at 4 sips of coffee instead of 0, so now it's obvious to me that if `ToAliasT` is identity except for tuples, then `ToAliasT (a, b) ~ (ToAliasT a, ToAliasT b) ~ (a, b)` :sweat_smile: 

Let's try deleting it?

Reply:
> OK, I'm now at 4 sips of coffee instead of 0, so now it's obvious to me that if ToAliasT is identity except for tuples, then ToAliasT (a, b) ~ (ToAliasT a, ToAliasT b) ~ (a, b)

Yeah, that's what I'd thought. Giving deleting a try now :)

Reply:
Done!